app/models/user.rb

```diff
@@ -6,4 +6,7 @@ class User < ApplicationRecord
 
   has_many :likes, dependent: :destroy
   has_many :liked_tweets, through: :likes, source: :tweet
+
```

Comment:
Giving this requirements:

We all know that in twitter we can create a tweet and like a tweet. But we can also follow users. So a User can follow other users and also have followers.

This exercise is more tricky because you will need to implement a `Follow` model that references **twice** the `User` model. One reference to say what is the id of the user who follows and another to say what is the id of the user that is being followed.

To help you get going:
- Create a `Follow` model + its migration. The model should have `follower_user_id` and `followed_user_id` as attributes.
- Create the `followed` and `follower` association in the `Follow` model. For example it should be possible to:
```ruby
follow = Follow.first
follow.followed # returns an instance of the User who is being followed
follow.follower # returns an instance of the User who is the follower
```

For the really tricky part 😁:
- Improve the `User` model by adding the `followers` and `followings` associations. For example:
```ruby
user_a = User.first
user_a.followers # returns an array of all users that follow user_a
user_a.followings # returns an array of all users that user_a follows

Is my implementation correct?

I've seen others have done it like this:

```
has_many :following, class_name: 'Follow', foreign_key: 'follower_id'
  has_many :following_users, through: :following, class_name: 'User', source: :follower

  has_many :followers, class_name: 'Follow', foreign_key: 'followed_id'
  has_many :followed_users, through: :followers, class_name: 'User', source: :followed
```
  
  Which is the best way? What is the difference?
  
  Thanks!